control/tests/timeresp_test.py

```diff
@@ -384,37 +406,32 @@ def test_auto_generated_time_vector(self):
             .1)
         np.testing.assert_array_almost_equal(
             _ideal_tfinal_and_dt(TransferFunction(1, [1, 2*zeta*wn, wn**2]))[1], 
-            1/(7.0*wn))
+            1/(ratio_dt*ratio*wn))
+        
+        
         # TF with fast oscillations simulates only 5000 time steps even with long tfinal
         self.assertEqual(5000, 
             len(_default_time_vector(TransferFunction(1, [1, 0, wn**2]),tfinal=100)))
-        # and simulates for 7.0/dt time steps 
-        self.assertEqual(
-            len(_default_time_vector(TransferFunction(1, [1, 0, wn**2]))), 
-            int(7.0/(1/(7.0*wn))))
         
         sys = TransferFunction(1, [1, .5, 0])
         sysdt = TransferFunction(1, [1, .5, 0], .1)
         # test impose number of time steps
         self.assertEqual(10, len(step_response(sys, T_num=10)[0]))
-        self.assertEqual(10, len(step_response(sysdt, T_num=10)[0]))
+        # test that discrete ignores T_num
+        self.assertNotEqual(15, len(step_response(sysdt, T_num=15)[0]))
         # test impose final time
         np.testing.assert_array_almost_equal(
             100, 
-            step_response(sys, 100)[0][-1], 
-            decimal=.5)
+            np.ceil(step_response(sys, 100)[0][-1]))
         np.testing.assert_array_almost_equal(
             100, 
-            step_response(sysdt, 100)[0][-1], 
-            decimal=.5)
+            np.ceil(step_response(sysdt, 100)[0][-1]))
         np.testing.assert_array_almost_equal(
             100, 
-            impulse_response(sys, 100)[0][-1], 
-            decimal=.5)
+            np.ceil(impulse_response(sys, 100)[0][-1]))
         np.testing.assert_array_almost_equal(
             100, 
-            initial_response(sys, 100)[0][-1], 
-            decimal=.5)
+            np.ceil(initial_response(sys, 100)[0][-1]))
```

Comment:
This is comparing scalar integers. A simple `assert 100 == np.ceil(step_response(sys, 100)[0][-1])`should suffice then.

Reply:
Sounds sane to me. I'll see if i can merge these changes (for some reason I dont see the button to do so at the moment)

Reply:
Nah, the display is only your current diff. My suggestion would be replace all the `assert_array_almost_equal`

Reply:
Makes sense. Tried to push a change this morning but git is giving me troubles again; once I figure out how to push again I'll see if I can take another stab at it.

Reply:
Let me know when you need help with git

Reply:
No, you can keep your PR branch for reference, or delete.

Merge `python-control/master` (not `origin`) into your local master and push that to your `origin` (`sawyerbfuller/master`). If you haven't added that remote yet:

```
git remote add upstream https://github.com/python-control/python-control.git
git fetch --all
```

(replace the HTTPS URL with the SSH one, to enable write access. Only for maintainers, and be careful not to mess with the upstream/master branch then)

Then merge and push to your `origin`:
```
git switch master
git merge upstream/master
git push
```

Reply:
If you delete your branch here then checkout back to `master` with 

```bash
git checkout master
git fetch upstream # gets the official repo status to your local 
git rebase upstream/master # applies the official master to your master
git push origin # sends it back to your own remote repo
```

then if you want you can delete your merged branch by first getting the acknowledgement that you already deleted the remote branch

```
git fetch -p # origin is implied when omitted
git branch -d better_final
```

Reply:
Thanks for the help guys. 

On github, it says my fork is eg. 40 commits behind python-control:master. When I do `git fetch --all`, I still don't get recent merged pull requests on python-control/python-control. Could it be because I have my remote configured incorrectly? 
 
When I first downloaded my local copy I did: 
``git clone https://github.com/sawyerbfuller/python-control.git``
and then
``git remote add upstream https://github.com/sawyerbfuller/python-control.git`` (my own fork)

If that's part of the problem, can I just change the remote e.g. with ``git remote set-url <remote_name> <remote_url>`` or should I re-download? Thank you!

Reply:
Hi, you just cought me before going to bed. Just a few remarks:

1. The way you describe it, you would have too remotes `origin` and  `upstream` pointing to the same URL. Use `set-url` to point `upstream` to python-control/python-control`
2. `git fetch --all` gets you the references but does not update your local branches. Only a `git merge upstream/master`will bring your local branch to the same state as the one `upstream`
3. Push to your origin to get your online branches on `origin`to the desired state. After that you are set to create new pull requests.

Reply:
OK looks like the problem was that I had set upstream to point to my own fork so I could never get updates from upstream.  `git remote set-url https://github.com/python-control/python-control.git` seems to have done the trick. Thanks!